src/domains/ecommerce-app/api/npg-notification/_npg_notifications_policy.xml.tpl

```diff
@@ -1,11 +1,43 @@
 <policies>
     <inbound>
         <base />
-        <!-- start policy variables -->
+        <!-- start validation policy -->
+        <set-variable name="orderId" value="@(context.Request.MatchedParameters["orderId"])" />
+        <validate-jwt query-param-name="sessionToken" failed-validation-httpcode="401" failed-validation-error-message="Unauthorized" require-expiration-time="true" require-signed-tokens="true" output-token-variable-name="jwtToken">
```

Comment:
Where is the token generated? because we could use a different key to generate the tokens for the `notificationUrl` than those used for the checkout/ecommerce payment flow.

Reply:
`ecommerce-jwt-signing-key` is used for payment-flow

Reply:
The token is generated in payment method or transactions service, i guess using the same key. 
I agree with you that the key for sure is different from the key used in checkout/ecommerce payment flow.
But do you agree with me that both `jwtToken` that have to passed to the `esito url ` generated from `paymentMethod` (for guest payment) and from `transaction service` (for logged payment) must be **signed** with the **same key**?

Reply:
yep